server/graphql/general/src/queries/discount.rs

```diff
@@ -0,0 +1,51 @@
+use async_graphql::*;
+use graphql_core::{
+    standard_graphql_error::{validate_auth, StandardGraphqlError},
+    ContextExt,
+};
+
+use service::{
+    auth::{Resource, ResourceAccessRequest},
+    pricing::discount::ItemDiscountLookup,
+};
+
+#[derive(InputObject, Clone)]
+pub struct ItemDiscountInput {
+    name_id: String,
+    item_id: String,
+}
+
+pub async fn sell_price_discount(
+    ctx: &Context<'_>,
+    store_id: String,
+    input: ItemDiscountInput,
+) -> Result<f64> {
+    let user = validate_auth(
+        ctx,
+        &ResourceAccessRequest {
+            resource: Resource::QueryMasterList, // Discount data comes from master list ...
+            store_id: Some(store_id.clone()),
+        },
+    )?;
+
+    let service_provider = ctx.service_provider();
+    let service_context = service_provider.context(store_id, user.user_id)?;
+
+    let discount = service_provider
+        .pricing_service
+        .get_discount_for_item_and_name_link_id(&service_context, input.to_domain())
+        .map_err(|e| StandardGraphqlError::from_repository_error(e))?;
+
+    Ok(discount)
+}
+
+impl ItemDiscountInput {
+    pub fn to_domain(self) -> ItemDiscountLookup {
+        let ItemDiscountInput { name_id, item_id } = self;
+
+        ItemDiscountLookup {
+            name_link_id: name_id,
```

Comment:
service layer shouldn't really be concerned with name_links, it happens in repos layer: https://github.com/msupply-foundation/open-msupply/blob/dd5ee23a3455d0ed786d68c7ae00439bd1720c5f/server/repository/src/db_diesel/master_list.rs#L93-L96

Reply:
Thanks! [d282d20](https://github.com/msupply-foundation/open-msupply/pull/4749/commits/d282d208ee44b28bd4931ad14542a78c0d7cd26c)